### DISCUSSION TITLE
PhotoSync Connection with TrueNAS Version

### DISCUSSION THREAD

**Discussion by InfinityRed-Code** [photoprism/photoprism#4415] [Q&A] [answered]
Hey,

this might be the wrong place to ask this and if it is I am sorry.
I have tried a lot of things now. My current setup is a freshly installed truenas on my Zima Blade where I made my mirror pool and installed PhotoPrism via the builtin App/Container feature. Everything works fine, except the connection to photosync on my phone, which was the goal in the first place. I am using the official, TrueNAS reccomended version of the container and I added two normal users. The Error I am getting in PhotoSync is a 404. Domain is correct, user and password are, I changed the port to 443 as I am hosting it behind a nginx proxy manager and I checked the SSL flag. Using the native admin user rsults in the same error, just like giving a optional target directory or not. I even once tried manually creating the directories and chmoding them. No change. I am on Android and the PhotoPrism Gallary App works fine, just like the rest.

Any help or a hint to where I can ask further would be highly apreciated.

**Comment by graciousgrey** [comment, marked as answer]
Hi :wave: 

You can find our troubleshooting checklist here:

https://docs.photoprism.app/getting-started/troubleshooting/#no-webdav-access

I would recommend going through it step by step. Also, if you have 2FA enabled for your user account, make sure you use an app password for webdav access.

https://docs.photoprism.app/user-guide/users/2fa/#step-3-app-passwords

**Reply by InfinityRed-Code** [reply]
OMG okay. Thank you. When Troubleshooting on my own I did not realise, that I have to manually enable webdav for users via cli aswell. I just saw that it is not disabled in the usersettings.

**Comment by InfinityRed-Code** [comment]
One last thing. I get permission errors when trying to upload anything. In the PhotoSync app the Error is 409 and in the Library Errors there are permission errors. I let the container create its own folders in TrueNAS, that I can avoid this issue. And the Installation of the app worked fine as well, so the permissions should work if I think about it. Does not wok in web as well and also not with the admin user.

**Reply by InfinityRed-Code** [reply]
@graciousgrey 
PS: I am not impatient, I just don't know weather you get a notification at all if I don't tag you.

**Reply by graciousgrey** [reply]
Your PhotoPrism instance requires write permissions for the configured import, originals and storage folders.

https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions

https://docs.photoprism.app/getting-started/docker-compose/#volumes

Unfortunately, we cannot help much with TrueNAS-specific problems, as this third-party integration is not maintained by us.

**Reply by InfinityRed-Code** [reply]
I had to also manually set permissions. Thank you
